### PR TITLE
Add a user-friendly diagnostic for compute-checksum command.

### PR DESCRIPTION
**Pull request by ryanwilson**
Starting as a draft while I build and verify locally that this works as expected and doesn't break anything else.

### CODE REVIEW

**Comment by ryanwilson** [comment]
That's a fantastic idea! Did you want me to change my commit or did you want to open a separate PR and I'll close this one? (only because I'd be deleting my lines and copying yours). I'm good with either, thanks!

**Comment by aciidb0mb3r** [comment]
@ryanwilson You should make the change and update this PR 😄. Please also add a corresponding unit test for it. Let us know if you need help figuring how to write a unit test for this change.

**Comment by ryanwilson** [comment]
@hartbit @aciidb0mb3r I modified the existing test that was expecting a diagnostic to pass, but not sure if you wanted me to add another one explicitly or not.

If so, I was thinking of:

1. Modifying `MockArchiver` to be able to pass in a `Set<String>` to override the `supportedExtensions` (using a default argument of `["zip"]`
2. Adding a test (likely in `WorkspaceTests.swift`?) below that uses a custom `MockArchiver` with multiple supported extensions, testing both ones that match and ones that do not.

I just realized now though: expecting the exact string may be an issue since `supportedExtensions` is a `Set`, meaning the order isn't necessarily guaranteed (or if it is, that's an implementation detail). Should I change the implementation to alphabetize the `Set` first before listing it?

**Comment by hartbit** [comment]
LGTM!I wouldn't bother to generate the supported extensions list in the test: I would just test against the diagnostic for have now (with zip). We can always update the text if and when we end up adding more extensions.

**Comment by hartbit** [comment]
@swift-ci smoke test

File: Sources/Workspace/Workspace.swift [hunk 1]
```diff
@@ -853,8 +853,9 @@ extension Workspace {
         forBinaryArtifactAt path: AbsolutePath,
         diagnostics: DiagnosticsEngine
     ) -> String {
-        guard fileSystem.isFile(path) else {
-            diagnostics.emit(error: "file not found at path: \(path.pathString)")
+        guard let pathExtension = path.extension, archiver.supportedExtensions.contains(pathExtension) else {
```
Comment (hartbit) [inline Sources/Workspace/Workspace.swift]: We should still check that the path is a file. Directories can have extensions.

Comment (ryanwilson) [reply Sources/Workspace/Workspace.swift]: Thanks! Re-added, also makes sense the path may not exist on disk.

**Comment by ryanwilson** [comment]
Added some more tests for a valid filepath but no file exists, as well as a valid filepath but it's a directory. Happy to refactor or remove some of the tests if desired 🙂

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by ryanwilson** [comment]
Happy to help! Noticed another test failing in a different dir (sorry for not testing it...). Should be fixed now I think.

**Comment by hartbit** [comment]
@swift-ci smoke test

**Comment by hartbit** [comment]
@ryanwilson can you have a look at the test failures? Thanks

**Comment by ryanwilson** [comment]
Ahh I misinterpreted what the test failure was. I added a `.zip` file of the
`.xcframework` in `Fixtures/BinaryTargets` and test that `zip` instead of the Markdown file. The `zip` file was created with `zip -r
MyFwk.zip MyFwk.xcframework`.

Thanks, all is passing locally for me 👍

**Comment by hartbit** [comment]
@swift-ci smoke test

**Comment by ryanwilson** [comment]
Thanks @hartbit and @aciidb0mb3r for the reviews and helping me get this merged 😄

**Pull request merged**